core/src/main/java/org/pentaho/platform/util/Emailer.java

```diff
@@ -240,12 +246,11 @@ public boolean send() {
     try {
       // Get a Session object
       Session session;
-
-      if ( authenticate ) {
-        session = Session.getInstance( props, authenticator );
-      } else {
-        session = Session.getInstance( props );
+      IEmailConfiguration emailConfig = null;
+      if ( service != null ) {
+        emailConfig = service.getEmailConfig();
       }
+      session = getSmtpSession( authenticate );
```

Comment:
misleading function name `getSmtpSession(` can return session that has nothing to do with smtp, right?
For example, in the case of Oauth?

Reply:
makes sense